service/history/shard/context.go

```diff
@@ -115,7 +115,7 @@ type (
 		GetWorkflowExecution(ctx context.Context, request *persistence.GetWorkflowExecutionRequest) (*persistence.GetWorkflowExecutionResponse, error)
 		// DeleteWorkflowExecution add task to delete visibility, current workflow execution, and deletes workflow execution.
 		// If branchToken != nil, then delete history also, otherwise leave history.
-		DeleteWorkflowExecution(ctx context.Context, workflowKey definition.WorkflowKey, branchToken []byte, closeExecutionVisibilityTaskID int64, stage *tasks.DeleteWorkflowExecutionStage) error
+		DeleteWorkflowExecution(ctx context.Context, workflowKey definition.WorkflowKey, branchToken []byte, closeExecutionVisibilityTaskID int64, workflowCloseTime time.Time, stage *tasks.DeleteWorkflowExecutionStage) error
```

Comment:
yea I added as `time.Time` because when we are creating visibility close task from proto, [`deleteVisibilityTask.CloseTime.AsTime()`](https://github.com/temporalio/temporal/pull/6168/files#diff-5c510ceac08a7fc24c1f03ee28f708af8ea8f0344cb1cd05b5e29fc4e6ae59f4:~:text=WorkflowCloseTime%3A%20%20%20%20%20%20%20%20%20%20%20%20%20%20deleteVisibilityTask.CloseTime.AsTime()%2C) always return `time.Time`. 

I also noticed that we are passing `closeExecutionVisibilityTaskID` as non-pointer object and we identify if it is "nil" by checking if that parameter equals to 0. So, I thought maybe we could do the similar thing for the `closetime` as well (i.e., identity if the closetime exists by checking if `task.WorkflowCloseTime.After(time.Unix(0, 0)`.